#OpenXT:Use WPA-PSK mechanism to authenticate on a WLAN connection.
--- a/configure.ac
+++ b/configure.ac
@@ -49,6 +49,10 @@ dnl Checks for typedefs, structures, and
 dnl
 AC_TYPE_PID_T
 
+dnl Add libcrypto for PKCS5_PBKDF2_HMAC_SHA1
+dnl OpenXT patch always-use-psk-hash.patch
+PKG_CHECK_MODULES(OPENSSL, libcrypto libssl,, AC_MSG_ERROR([openssl cannot be found.]))
+
 dnl
 dnl translation support
 dnl
--- a/Makefile.am
+++ b/Makefile.am
@@ -270,12 +270,14 @@ src_wireless_security_libwireless_securi
 	$(dflt_cppflags) \
 	"-I$(srcdir)/shared" \
 	$(GTK3_CFLAGS) \
-	$(LIBNM_CFLAGS)
+	$(LIBNM_CFLAGS) \
+	$(OPENSSL_CFLAGS)
 
 src_wireless_security_libwireless_security_libnm_la_LIBADD = \
 	src/utils/libutils-libnm.la \
 	$(GTK3_LIBS) \
-	$(LIBNM_LIBS)
+	$(LIBNM_LIBS) \
+	$(OPENSSL_LIBS)
 
 EXTRA_DIST += \
 	src/wireless-security/meson.build
@@ -396,7 +398,8 @@ src_connection_editor_nm_connection_edit
 	$(LIBNMA_CFLAGS) \
 	$(LIBNM_CFLAGS) \
 	$(JANSSON_CFLAGS) \
-	$(SELINUX_CFLAGS)
+	$(SELINUX_CFLAGS) \
+	$(OPENSSL_CFLAGS)
 
 src_connection_editor_nm_connection_editor_LDADD = \
 	src/wireless-security/libwireless-security-libnm.la \
@@ -405,6 +408,7 @@ src_connection_editor_nm_connection_edit
 	$(LIBNMA_LIBS) \
 	$(JANSSON_LIBS) \
 	$(SELINUX_LIBS) \
+	$(OPENSSL_LIBS) \
 	-lm
 
 $(src_connection_editor_nm_connection_editor_OBJECTS): $(connection_editor_h_gen)
@@ -527,7 +531,8 @@ src_nm_applet_CPPFLAGS = \
 	$(LIBSECRET_CFLAGS) \
 	$(NOTIFY_CFLAGS) \
 	$(MM_GLIB_CFLAGS) \
-	$(APPINDICATOR_CFLAGS)
+	$(APPINDICATOR_CFLAGS) \
+        $(OPENSSL_CFLAGS)
 
 src_nm_applet_LDADD = \
 	src/wireless-security/libwireless-security-libnm.la \
@@ -538,6 +543,7 @@ src_nm_applet_LDADD = \
 	$(NOTIFY_LIBS) \
 	$(MM_GLIB_LIBS) \
 	$(APPINDICATOR_LIBS) \
+	$(OPENSSL_LIBS) \
 	-lm
 
 $(src_nm_applet_OBJECTS): $(nm_applet_h_gen)
